### DISCUSSION TITLE
[Plugin]: 404 error

### DISCUSSION THREAD

**Discussion by AgedLace** [ObsidianPublisher/obsidian-github-publisher#194] [Q&A — Plugin] [answered]
### Issue validation

- [X] I checked the issue to prevent duplicate
- [X] I checked my configurations files and the documentation

### Command used

Upload single current active note

### Plugin version

6.4.0

### Describe the bug

Run actions/configure-pages@v2
  with:
    token: ***
    enablement: true
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.4/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib
Warning: Get Pages site failed
Error: Create Pages site failed
Error: AxiosError: Request failed with status code 403

### How to reproduce ?

I have only one file ... index.md [copy and pasted from above].  When I use the Command Palette to push the single file, I get the above error.

---

Run actions/configure-pages@v2
  with:
    token: ***
    enablement: true
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.4/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.4/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib
Warning: Get Pages site failed
Error: Create Pages site failed
Error: AxiosError: Request failed with status code 403

### Minimal Reproducible Example

my index.md file

---
title: "Test"
share: true
---
# Test
Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.

### Configuration

```JSON
{
  "github": {
    "branch": "main",
    "automaticallyMergePR": true,
    "api": {
      "tiersForApi": "Github Free/Pro/Team (default)",
      "hostname": ""
    },
    "workflow": {
      "name": "",
      "commitMessage": "[skip netlify] PUBLISHER"
    },
    "tokenPath": ".obsidian/plugins/obsidian-mkdocs-publisher/env"
  },
  "upload": {
    "behavior": "yaml",
    "subFolder": "",
    "defaultName": "docs/hidden",
    "rootFolder": "docs",
    "yamlFolderKey": "category",
    "frontmatterTitle": {
      "enable": true,
      "key": "name"
    },
    "replaceTitle": [],
    "autoclean": {
      "enable": true,
      "excluded": [
        "docs/assets/js",
        "docs/assets/meta",
        "docs/assets/css",
        "tags.md",
        "graph.md"
      ]
    },
    "folderNote": {
      "enable": true,
      "rename": "index.md"
    },
    "metadataExtractorPath": "",
    "replacePath": []
  },
  "conversion": {
    "hardbreak": false,
    "dataview": false,
    "censorText": [],
    "tags": {
      "inline": true,
      "exclude": [],
      "fields": []
    },
    "links": {
      "internal": true,
      "unshared": false,
      "wiki": false
    }
  },
  "embed": {
    "attachments": true,
    "keySendFile": [
      "image"
    ],
    "notes": true,
    "folder": "docs/assets/img"
  }
}
```

### Relevant log output

```bash session
I have no idea what is "relevant".  Copying the whole thing.

﻿
plugin:url-into-selection:690 loading url-into-selection
plugin:quickadd:19480 Loading QuickAdd
plugin:quickadd:17480 QuickAdd: (LOG) No migrations to run.
 loading plugin "tasks"
plugin:obsidian-vaul…atistics-plugin:691 Loading vault-statistics Plugin
plugin:obsidian-outliner:3317 Loading obsidian-outliner
plugin:table-editor-obsidian:28663 loading markdown-table-editor plugin
plugin:obsidian-bett…-command-palette:15 Loading plugin: Better Command Palette
plugin:obsidian42-strange-new-worlds:13 loading Obsidian42 - Strange New Worlds (SNW)
plugin:callout-manager:1620 Unable to fetch Obsidian stylesheet. Error: [obsidian-extra]: Obsidian 1.3.7 has not been tested with this function
    at eval (plugin:callout-manager:154:13)
    at Generator.next (<anonymous>)
    at eval (plugin:callout-manager:147:67)
    at new Promise (<anonymous>)
    at __awaiter (plugin:callout-manager:129:10)
    at fetchObsidianStyles (plugin:callout-manager:152:10)
    at StylesheetWatcher.checkForChangesObsidian (plugin:callout-manager:1612:35)
    at StylesheetWatcher.checkForChanges (plugin:callout-manager:1594:28)
    at StylesheetWatcher.watch (plugin:callout-manager:1531:10)
    at eval (plugin:callout-manager:2983:37)
plugin:dataview:12575 Dataview: all 0 files have been indexed in 0.026s (0 cached, 0 skipped).
app.js:1 Uncaught RangeError: Field is not present in this state
    at e.field (app.js:1:338037)
    at getEditorFromState (plugin:obsidian-outliner:88:30)
    at VerticalLinesPluginValue.waitForEditor (plugin:obsidian-outliner:2552:28)
plugin:dataview:12579 Dataview: Dropped cache entries for 3 deleted files.
 Obsidian Admonition unloaded
 Obsidian Admonition loaded
 unloading plugin "tasks"
plugin:obsidian-tasks-plugin:254 loading plugin "tasks"
app.js:1 Plugin Update Tracker: styles.css not found
 Obsidian Admonition unloaded
plugin:obsidian-mkdocs-publisher:59 [GITHUB PUBLISHER] v.6.4.0 (lang: en) loaded
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] Moving the GitHub Token in the file : .obsidian/plugins/obsidian-mkdocs-publisher/env
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] Repository AgedLace/Bible-Study-Notebook exists. Now testing the {{- main}} branch.
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] You are not rate limited. You can make 4987 requests before the limit is reset at 7:34:37 PM.
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] Repository AgedLace/Bible-Study-Notebook exists. Now testing the {{- main}} branch.
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] Branch successfully created (status: 201) on AgedLace/Bible-Study-Notebook
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] Publishing file: Bible-Study-Notebook/docs/index.md
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] Publishing index.md to [AgedLace/Bible-Study-Notebook/main] 
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] Upload index.md:docs/hidden/index.md on AgedLace/Bible-Study-Notebook:Bible-Study-Notebook-8-13-2023
api.github.com/repos…otebook-8-13-2023:1 
 Failed to load resource: the server responded with a status of 404 ()
plugin:obsidian-mkdocs-publisher:20 [GITHUB PUBLISHER] The 404 error is normal ! It means that the file does not exist yet. Don't worry ❤️.
﻿
```

### OS

Linux

### Anything else?

My brand new vault only has the one file.  The one you designated as "test".  

### Obsidian information

```bash session
SYSTEM INFO:
	Obsidian version: v1.3.7
	Installer version: v1.3.7
	Operating system: #1 SMP PREEMPT_DYNAMIC Wed Aug  9 09:02:26 UTC 2023 6.1.44-1-MANJARO
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: Dracula Plus v1.3.5
	Snippets enabled: 9
	Restricted mode: off
	Plugins installed: 45
	Plugins enabled: 32
		1: Better Inline Fields v0.3.5
		2: Callout Manager v1.0.1
		3: Code Block Enhancer v1.0.5
		4: Dataview v0.5.56
		5: Drag-n-Drop for blocks v1.2.0
		6: Editing Toolbar v2.3.1
		7: Editor Syntax Highlight v0.1.3
		8: Find orphaned files and broken links v1.9.0
		9: Highlightr v1.2.2
		10: Janitor v1.0.7
		11: Multi-Column Markdown v0.8.3
		12: Natural Language Dates v0.6.1
		13: Note Refactor v1.7.1
		14: Obsidian Tabs v1.0.7
		15: Paste URL into selection v1.7.0
		16: QuickAdd v1.2.1
		17: Style Settings v1.0.6
		18: Tag Wrangler v0.5.11
		19: Templater v1.16.0
		20: Vault Statistics v0.1.3
		21: Outliner v4.8.0
		22: Advanced Tables v0.19.1
		23: Zotero Integration v3.0.7
		24: Better Command Palette v0.15.0
		25: Annotator v0.2.10
		26: Advanced Random Note v0.0.7
		27: Obsidian42 - Strange New Worlds (SNW) v1.2.2
		28: Omnivore v1.6.0
		29: Tasks v4.6.0
		30: Plugin Update Tracker v1.5.2
		31: Wordy v1.3.1
		32: Github Publisher v6.4.0

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.
```

**Comment by Mara-Li** [comment]
Could you try to use the verification repository command's first?

There is a button in the setting for that.

**Comment by AgedLace** [comment]
I'm sorry, I answered this via email. I didn't know I should have responded here.  I've never submitted this type of thing before.  
Here is what I sent last night ...

---

Yes, I did that right after I had set up the plugin with my github credentials and Github token.  It said everything appeared to be valid. 

I just double checked it again.  It said the same. 

I issued the publish single active note again.  This time I got a message saying that the upload was successful.

In my Actions tab on Github, the "pages build and deployment" passed, but there is no creation of the gh-pages branch.  There is only the 'main' branch.

Thank you for your assistance.

---

Again, I'm very sorry for any inconvenience!!!

**Reply by Mara-Li** [reply]
Yeah it's normal you don't have gh-branch ! It was deprecated by GitHub, and I created a special workflow that don't use it! What workflow and template do you use ? 
Here the file you must check:

- [ ] The file in `.obsidian/plugins/obsidian-mkdocs-publisher/env` have the token in it and is not empty? Sometimes it bugs (dunno why) and get emptier, mainly because of sync. 
- [ ] The file is successfully in your repository 
- [ ] the workflow as no error
- [ ] Finally, check the settings of Page (https://obsidian-publisher.netlify.app/template/publishing/#through-github-pages). You need to activate "Github workflow", run it and wait to the page to appear. 

BTW, no need to worry!

**Comment by AgedLace** [comment]
This is the template I am using ...

https://github.com/ObsidianPublisher/template-gh-pages

I do not know which workflow is being used.  There are 4 of them in the .github/workflows folder. They are ...

deploy.yml
index.yml
maintenance.yml
update.yml

On the "Actions" Tab, it says this ... [skip netlify] PUBLISHER (commit number)


The .obsidian/plugins/obsidian-mkdocs-publisher/env is not empty; and it has the correct Token.  (I just double checked it to make sure).

The file is NOT being uploaded into my repository.  In Obsidian, I used the Command Palette to send "Upload single current active note".  After several seconds, I get the popup message / notification that the file has been successfully sent to the repo.  However, this is not the case.

The workflow has no error

I fixed the Github workflow setting; it was unchecked.

AFTER I fixed this, I re-ran the Command Palette Command with the same result.

I also double checked that my Source is in fact Github Actions.

**Comment by AgedLace** [comment]
Correction to the above ...

in the "Actions" tab there are two workflows being processed.

The first one is "pages build and deployment" - no errors
The second one is the [skip netlify] PUBLISHER (commit number) - again, no errors.

**Reply by Mara-Li** [reply]
Okay, so the file is not send!

**Reply by Mara-Li** [reply]
As there are some 404 error, did you check if the repository and the username are correct ?

**Reply by Mara-Li** [reply]
Can I get the "github" part of the `data.json` ?

**Comment by AgedLace** [comment]
Correct.  I also double-checked to ensure that I placed it in the 'docs' folder.  That is where it is located in my fault.

Rechecked repo credentials ... Repo name is "Bible-Study-Notebook"
User Name is "AgedLace.
Repo is here ... https://github.com/AgedLace/Bible-Study-Notebook

I'm not finding data.json?  I'm looking in .obsidian directory

**Reply by Mara-Li** [reply]
It's in `.obsidian/plugins/obsidian-mkdocs-publisher` !

**Comment by AgedLace** [comment]
Github portion of data.json
---

{
  "github": {
    "branch": "main",
    "automaticallyMergePR": true,
    "api": {
      "tiersForApi": "Github Free/Pro/Team (default)",
      "hostname": ""
    },

**Reply by Mara-Li** [reply]
there is no username …? The export usually delete it, and I need it!

**Comment by AgedLace** [comment]
I copy and pasted it as it is in my data.json file?

**Reply by Mara-Li** [reply]
Mine looks like this: 
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/30244939/7d669edc-f836-4098-ae3c-f36fde7444c1)

**Comment by AgedLace** [comment]
Okay. I'll edit mine to conform to yours (using my info).  Give me a few.

**Comment by AgedLace** [comment]
AHHHH ... in my data.json ... my repo and user are under "tokenPath".  Let me paste the entire contents of the file.
---
{
  "github": {
    "branch": "main",
    "automaticallyMergePR": true,
    "api": {
      "tiersForApi": "Github Free/Pro/Team (default)",
      "hostname": ""
    },
    "otherRepo": [],
    "workflow": {
      "name": "",
      "commitMessage": "[skip netlify] PUBLISHER"
    },
    "tokenPath": ".obsidian/plugins/obsidian-mkdocs-publisher/env",
    "repo": "Bible-Study-Notebook",
    "user": "AgedLace",
    "rateLimit": 4998,
    "verifiedRepo": true
  },
  "upload": {
    "behavior": "yaml",
    "subFolder": "",
    "defaultName": "docs/hidden",
    "rootFolder": "docs",
    "yamlFolderKey": "category",
    "frontmatterTitle": {
      "enable": true,
      "key": "name"
    },
    "replaceTitle": [],
    "autoclean": {
      "enable": true,
      "excluded": [
        "docs/assets/js",
        "docs/assets/meta",
        "docs/assets/css",
        "tags.md",
        "graph.md"
      ]
    },
    "folderNote": {
      "enable": true,
      "rename": "index.md"
    },
    "metadataExtractorPath": "",
    "replacePath": []
  },
  "conversion": {
    "hardbreak": false,
    "dataview": false,
    "censorText": [],
    "tags": {
      "inline": true,
      "exclude": [],
      "fields": []
    },
    "links": {
      "internal": true,
      "unshared": false,
      "wiki": false
    }
  },
  "embed": {
    "attachments": true,
    "keySendFile": [
      "image"
    ],
    "notes": true,
    "folder": "docs/assets/img"
  },
  "plugin": {
    "shareKey": "share",
    "fileMenu": false,
    "editorMenu": false,
    "excludedFolder": [],
    "copyLink": {
      "enable": false,
      "links": "",
      "removePart": [],
      "addCmd": false
    },
    "noticeError": false,
    "displayModalRepoEditing": false
  }
}

**Comment by Mara-Li** [comment]
OKAY
So, I checked the repo.

The file are in `docs/hidden` (as normally should, using your settings!)
http://benotdeceived.org/Bible-Study-Notebook/ are not normally build, so there is seems to be a problem around mkdocs or the `mkdocs.yml` file!

https://github.com/AgedLace/Bible-Study-Notebook/blob/main/mkdocs.yml You need to update it !!! That's why you get not the good things :>

**Comment by AgedLace** [comment]
I don't know how to update that file?

**Reply by Mara-Li** [reply]
Oh, I'm sorry! You seems new to GitHub ?
Well, maybe this guide can help you : https://github.com/skills/introduction-to-github (more here: https://skills.github.com/)

**Comment by AgedLace** [comment]
Ohhh ... do you mean to set my site name to https://BeNotDeceived.org/Bible-Study-Notebook?

**Reply by Mara-Li** [reply]
```yaml
site_name: your_site_name # Change this to your site name
site_description: A cool description of your site
site_url: https://change_me.io/ # Change this to your site URL
```

Change each part as you want, but the URL MUST be `http://benotdeceived.org/Bible-Study-Notebook/`

**Comment by AgedLace** [comment]
Okay ... the file was indeed uploaded to docs/hidden.

I don't understand, because I set "share" true".  Should it not just be "docs/index.md" ?

**Reply by Mara-Li** [reply, marked as answer]
No, because you set the default folder to `docs/hidden`

**Comment by AgedLace** [comment]
Ahhh ... I used the "mkdocs" presets.  Let me change that.

**Comment by AgedLace** [comment]
Success!!! https://benotdeceived.org/Bible-Study-Notebook/

THANK YOU so very much for your time!! I VERY MUCH appreciate it!!

**Reply by Mara-Li** [reply]
Ty for your kindness! Happy to help! I hope it will help more people :>

**Comment by AgedLace** [comment]
Not sure if you got the email last night.  Will add it here, too.

---

I just received notification that the update.yml workflow failed. (update/get-env)

I've noticed that the repo's name is doubled in the following links in the error messages in my previous email.

---

Downloading file: update.txt to: /home/runner/work/Bible-Study-Notebook/Bible-Study-Notebook
Done: /home/runner/work/Bible-Study-Notebook/Bible-Study-Notebook/update.txt

---

I don't know how to fix this?

Thank you so much for your assistance!

**Reply by Mara-Li** [reply]
Hmm, it's probably on my side for this.
I will check your error and try to fix the bug :).

**Reply by Mara-Li** [reply]
You can normally rerun the actions ! 
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/30244939/728a7f38-a406-447b-9188-8df4d826191d)
Just click on the button "run workflow"

**Comment by AgedLace** [comment]
Received the following error message on manually rerunning the action.

---

0s
0s
Run actions-js/push@master
Started: bash /home/runner/work/_actions/actions-js/push/master/start.sh
Push to branch main
Missing input "github_token: ${{ secrets.GITHUB_TOKEN }}".
Error: Invalid status code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/actions-js/push/master/start.js:9:19)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 1
}
Error: Invalid status code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/actions-js/push/master/start.js:9:19)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)

**Reply by Mara-Li** [reply]
Yeah, you need to set a Github Token in your secrets!
https://obsidian-publisher.netlify.app/template/actions/#update-updateyml

**Comment by AgedLace** [comment]
I have created the secret Github Token and re-ran the workflow.

It errored out again - saying the Secret Token was missing.  Went and confirmed that the Secret was registered.  It says the my repository secret was updated 5 minutes ago.

Re-ran the workflow a second time, and it still fails for missing secret Github Token.

---

Run actions-js/push@master
Started: bash /home/runner/work/_actions/actions-js/push/master/start.sh
Push to branch main
Missing input "github_token: ${{ secrets.GITHUB_TOKEN }}".
Error: Invalid status code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/actions-js/push/master/start.js:9:19)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 1
}
Error: Invalid status code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/actions-js/push/master/start.js:9:19)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)

**Reply by Mara-Li** [reply]
Did you name it "GH_TOKEN" ?

**Comment by AgedLace** [comment]
I renamed it to GH_TOKEN, and reran the workflow.  Now there is a new error.

---

Run peter-evans/create-pull-request@v4
(node:1874) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Save persisted git credentials
Determining the base and head repositories
Pull request branch target repository set to AgedLace/Bible-Study-Notebook
Configuring credential for HTTPS authentication
Checking the base repository state
Restore persisted git credentials

**Reply by Mara-Li** [reply]
Seems to be a problem caused by peter-evans/create-pull-request@v4 :/
Need to check the repository and set a Issue :/
Try to set the inputs for auto-merging PR to true ?

**Reply by Mara-Li** [reply]
EDIT: I updated the version of the workflow, you can retry

**Comment by AgedLace** [comment]
Under "Create pull request" get this error.

---

0s
Run peter-evans/create-pull-request@v5
Prepare git configuration
Determining the base and head repositories
Pull request branch target repository set to AgedLace/Bible-Study-Notebook
Configuring credential for HTTPS authentication
Checking the base repository state
Restore git configuration

-

**Comment by AgedLace** [comment]
Sorry, I keep forgetting to open each step ... here is where the error is happening ...

---

Checking the base repository state
  /usr/bin/git symbolic-ref HEAD --short
  main
  Working base is branch 'main'
  /usr/bin/git remote prune origin
  fatal: could not read Username for 'https://github.com': No such device or address
  Error: The process '/usr/bin/git' failed with exit code 128

**Reply by Mara-Li** [reply]
I'm sorry, I don't know :/

**Comment by AgedLace** [comment]
Okay.  

I'm sorry I've caused so many issues.  I'll have to look for a different option.

THANK YOU for your patience and your assistance.  It is still *VERY MUCH* appreciated.

**Reply by Mara-Li** [reply]
Checked the repository, and some people have this error: https://github.com/peter-evans/create-pull-request/issues/2094
Maybe check that/open an issue here!

**Comment by AgedLace** [comment]
THANKS!  Will definitely check that out!
